frontend/ui/impl/src/main/java/com/virtuslab/gitmachete/frontend/ui/impl/table/GitMacheteRepositoryDiscoverer.java

```diff
@@ -57,7 +58,8 @@ public void enqueue(Path macheteFilePath) {
       public void run(ProgressIndicator indicator) {
         LOG.debug("Running automatic discover task");
         val gitMacheteRepositoryInstantiateResult = Try
-            .of(() -> RuntimeBinding.instantiateSoleImplementingClass(IGitMacheteRepositoryCache.class)
+            .of(() -> ApplicationManager.getApplication()
```

Comment:
Hmmm actually a high-level question is... does this transition from _poor man's dependency injection_ to _poor man's DI but wrapped into ApplicationService_ bring any value? the downsides include one extra layer of indirection + one extra way of DI (`RuntimeBinding`, `@Service`, now also `ApplicationService`)

Reply:
I can't see any real value in it, the code is quite similar. There's an IntelliJ guarantee that for calls from multiple threads the service will be initialized in the first one, but the implementation from before doesn't have that problem at all.

Reply:
Okay... so unless @mkondratek insists otherwise, pls crop this PR to just include renames

Reply:
Done

Reply:
this is fine

Reply:
_this is fine_ or `:this_is_fine:`?